### PR TITLE
[diags] add minimal IEEE 802.15.4 frame size check

### PR DESCRIPTION
**Pull request by canisLupus1313**
Extending checks done by diag send command and diag repeat command
to validate minimal size of frame.

### CODE REVIEW

**Review by zhanglongxia — APPROVED** [review]
LGTM

File: src/core/diags/README.md [hunk 1]
```diff
@@ -76,6 +76,7 @@ status 0x00
 ### diag send \<packets\> \<length\>
 
 Transmit a fixed number of packets with fixed length.
+Length parameter has to be in range [3;127].
```
Comment (edmont) [inline src/core/diags/README.md:79]: Comma is used instead of semicolon for expressing ranges in this file.

**Review by edmont — APPROVED** [review]
LGTM.

File: src/core/diags/README.md (hunk 1 above)
Comment (canisLupus1313) [reply src/core/diags/README.md]: Fixed.

**Comment by canisLupus1313** [comment]
@zhanglongxia Could You rettriger the workflows?

**Comment by zhanglongxia** [comment]
> @zhanglongxia Could You rettriger the workflows?

Could you fix the pretty build errors?

File: src/core/diags/factory_diags.cpp [hunk 2]
```diff
@@ -244,6 +244,7 @@ Error Diags::ProcessRepeat(uint8_t aArgsLength, char *aArgs[], char *aOutput, si
 
         SuccessOrExit(error = ParseLong(aArgs[1], value));
         VerifyOrExit(value <= OT_RADIO_FRAME_MAX_SIZE, error = kErrorInvalidArgs);
+        VerifyOrExit(value >= (OT_RADIO_FRAME_MIN_SIZE), error = kErrorInvalidArgs);
```
Comment (jwhui) [inline src/core/diags/factory_diags.cpp:247]: nit: to be consistent with line above
```suggestion
        VerifyOrExit(value >= OT_RADIO_FRAME_MIN_SIZE, error = kErrorInvalidArgs);
```

File: src/core/diags/factory_diags.cpp [hunk 3]
```diff
@@ -271,6 +272,7 @@ Error Diags::ProcessSend(uint8_t aArgsLength, char *aArgs[], char *aOutput, size
 
     SuccessOrExit(error = ParseLong(aArgs[1], value));
     VerifyOrExit(value <= OT_RADIO_FRAME_MAX_SIZE, error = kErrorInvalidArgs);
+    VerifyOrExit(value >= (OT_RADIO_FRAME_MIN_SIZE), error = kErrorInvalidArgs);
```
Comment (jwhui) [inline src/core/diags/factory_diags.cpp:275]: nit: to be consistent with line above
```suggestion
    VerifyOrExit(value >= OT_RADIO_FRAME_MIN_SIZE, error = kErrorInvalidArgs);
```

**Review by abtink — APPROVED** [review]
👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**